### DISCUSSION TITLE
How to setup HTTPS for Remix App Server?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [remix-run/remix#3831] [Q&A] [answered]
I'm currently using the Remix App Server template & self hosting the remix app at work/

I don't see an entry point where I could pass my credentials, similar to Express or Fastify
- https://www.fastify.io/docs/latest/Reference/HTTP2/
- http://expressjs.com/en/5x/api.html#app.listen

How would I setup https  (pass key & cert) for a self hosted Remix app? 

Hopefully I don't need to eject or change my template and eject out of `Remix App Server` 😅

**Comment by jgb-solutions** [comment]
Remix needs to be behind a proxy server such as NginX. The node server will start on a port you defined or the default. And then you use a reverse proxy like NginX to forward the requests on port 80 and 443 to your application. So the reverse proxy would handle your certs instead of the Remix server.

**Reply by cliffordfajardo** [reply]
That's definitely sounds like the best approach;  I gotta see how our infra is setup at work 🤞

**Comment by cliffordfajardo** [comment, marked as answer]
Dropping another answer from Remix Discord:
> Remix App Server is intended to be minimal, to the point of not supporting many basic things. If you can't do it in entry.server.ts or your loaders you probably need to upgrade to something else. https://github.com/remix-run/remix/discussions/3244 is a relevant discussion
> I recommend switching to express which is what remix-serve uses - should be an easy migration, either by manually building a server.js file that looks like https://github.com/remix-run/remix/blob/main/packages/remix-serve/index.ts or by using a script like https://github.com/kiliman/rmx-cli#eject-ras to eject
